docs/source/transaction_family_specification.rst

```diff
@@ -0,0 +1,149 @@
+***********************************************
+OMI Summer Lab Transaction Family Specification
+***********************************************
+
+Overview
+=========
+The transaction family for the Open Music Initiative Summer Lab is intended to
+lay a foundation for the management of identities, musical works, and recordings
+on the Sawtooth blockchain. It is expected that Summer Lab projects will use,
+extend, or even replace portions of this design in their projects and demos.
+
+The transaction family allows users to set and update:
+
+* Individual Identities - Songwriters, performers, DJs, etc.
+* Organizational Identities - Labels, Publishers, etc.
+* Musical Works - The composition of a song or piece of music.
+* Musical or Video Recordings - A sound or video recording of a music performance
+  or a recording derived from a number of recorded music performances.
+
+An OMI Summer Lab transaction request is defined by the following values:
+
+* A verb which describes what action the transaction takes (e.g. SetRecording)
+* An object which represents the full replacement value compatible with the action.
+
+For example, a 'SetRecording' transaction must contain a valid 'Recording' object.
+
+For simplicity, the initial OMI Summer Lab transaction family combines 'create' and
+'update' semantics into a single 'set' concept. If the unique identifier used to
+encode the address of the object does not exist in state, then the signer of the
+transaction becomes the 'registering owner' of that object. Any future 'set' actions
+against that object must be signed by the same private key as was used to perform the
+first 'set'. Each tranasaction also includes a full replace of the existing data.
+
+The initial actions are:
+
+* SetIndividualIdentity
+* SetOrganizationalIdentity
+* SetWork
+* SetRecording
+
+State
+=====
+The four different types of objects each have their own storage formats in global state,
+defined by Google Protocol Buffers v3 schemas.
+
+.. literalinclude:: ../../protos/identity.proto
+   :language: protobuf
+   :caption: File: protos/identity.proto
+   :linenos:
+
+.. literalinclude:: ../../protos/work.proto
+   :language: protobuf
+   :caption: File: protos/work.proto
+   :linenos:
+
+.. literalinclude:: ../../protos/recording.proto
+   :language: protobuf
+   :caption: File: protos/recording.proto
+   :linenos:
+
+Addressing
+----------
+OMI Summer Lab data is stored in state using addresses which are generated from the OMI
+Summer Lab namespace prefix, the object type prefix, and the unique key field of the object.
+
+Addresses will adhere to the following format:
+
+- Addresses must be a 70 character hexadecimal string
+- The first 6 characters of the address are the first 6 hexdigest characters of a sha512 hash of
+  the utf-8 formatted string 'OMI'
+- The next 2 characters of the address are based on the object type:
+
+  - '00' - IndividualIdentity
```

Comment:
What's the reasoning for these character choices? Can we make them mnemonically easier?